lib/phobos/actions/process_message.rb

```diff
@@ -57,10 +57,15 @@ def force_encoding(value)
       def process_message(payload)
         instrument('listener.process_message', @metadata) do
           handler = @listener.handler_class.new
-          preprocessed_payload = handler.before_consume(payload)
+          preprocessed_payload = handler.before_consume(payload, @metadata)
+          consume_block = Proc.new { handler.consume(preprocessed_payload, @metadata) }
 
-          @listener.handler_class.around_consume(preprocessed_payload, @metadata) do
-            handler.consume(preprocessed_payload, @metadata)
+          if @listener.handler_class.respond_to?(:around_consume)
+            # around_consume class method implementation
```

Comment:
Maybe emit some kind of deprecation, such as

```
warn Kernel.caller.first + " around_consume has been moved to instance method, please update your consumer. This will not be backwards compatible in the future."
```

I am not sure about how to best emit deprecation warnings or what the message should say, just a suggestion.